### PR TITLE
Mermaid integration

### PR DESCRIPTION
**Pull request by ptrelease**
Added Mermaid integration and a proposed data model that would support new business cases moving forward.

### CODE REVIEW

File: source/documentation/design/data_model.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+# Proposed data model
+
+This is data model we will aim for.  Its designed to accommodate new business
+cases as quickly and easily as possible.
+
+<%= partial 'partials/mermaid' %>
+
+<div class="mermaid">
+classDiagram
```
Comment (stuartmccoll) [inline source/documentation/design/data_model.html.md.erb:14]: Are the cardinalities linking to the `address_link` correct? Should it be *one* `person`, `organisation` or `application` can have *one to many* `address_link` items?

File: source/documentation/design/data_model.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  address_link ..> address : 1 to 1..*
+  application ..> address_link : 1 to 1
+  grant_programme_sff_os ..> application : 0..1 to 1
+  person ..> address_link : 1 to 1
+  organisation ..> address_link : 1 to 1
+  user ..> user_organisation : 1 to 1..*
+  user_organisation ..> organisation : 1 to 1
+  organisation ..> organisation_type: 1 to 1
+  organisation_mission ..> organisation: 1..* to 1
+  user ..> person : 1 to 1
```
[the remark this answers is not in the document]
Comment (ptrelease) [reply source/documentation/design/data_model.html.md.erb:25]: Good spot

File: source/documentation/design/data_model.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  mission ..> organisation_mission : 0..* to 1
+  outcome ..> outcome_type : 1 to 1
+  outcome ..> application : 1..* to 1
+  application ..> organisation : 1..* to 1
```
[the remark this answers is not in the document]
Comment (ptrelease) [reply source/documentation/design/data_model.html.md.erb:29]: Yep makes sense.

File: source/documentation/design/data_model.html.md.erb (hunk 1 above)
Comment (ptrelease) [reply source/documentation/design/data_model.html.md.erb:14]: Yes!

**Pull request merged**